### ISSUE TITLE
Docs on `remove`

### ISSUE THREAD

**Issue by daninus14** [fukamachi/qlot#224]
Using `qlot remove` I get `Command not found: remove`.

Can you please add docs on how to remove a system

- If the system was installed via quicklisp, should we just do `(ql:uninstall)`?
- If the systems was installed via `qlot add` or editing the `qlfile` and `qlot install` should we delete the folder for that system under `.qlot/dists/`?

I just had a bug where I added an already installed system via quicklisp with qlot (because of a fork for that system solving a certain bug) and slime would only load the quicklisp version. After doing `(ql:uninstall)`, the project wouldn't build even though the qlot version was in the directory.

I solved the issue by deleting both the `.qlot/dists/system-name` directory for the given `system-name` system, and `.qlot/dists/quicklisp/software/system-name`, and then running `qlot install` again.

I can do a pull request for the docs if you just confirm what the right approach is. I think: 

- deleting the `remove` command from [this section](https://github.com/fukamachi/qlot?tab=readme-ov-file#remove)
- add a new "How to Remove a Dependency" in the [tutorial section](https://github.com/fukamachi/qlot?tab=readme-ov-file#tutorial)

The question is how to actually do it? Is it by deleting both directories? or `(ql:uninstall)`?

**Comment by fukamachi** [comment]
Which version of Qlot are you using?

**Comment by daninus14** [comment]
```
$ qlot --version
Qlot 1.2.11
```

**Comment by fukamachi** [comment]
Okay, `qlot remove` was added in [v1.3.0](https://github.com/fukamachi/qlot/releases/tag/1.3.0).

In v1.2.x, delete a line in qlfile and run `qlot install`.

**Comment by daninus14** [comment]
Thanks for the info. Hm, I ran `ros update qlot` before I posted that to check, but still got 1.2.x.

Here's the output
```
$ ros update qlot
git pull on /Users/username/.roswell/local-projects/fukamachi/qlot/
remote: Enumerating objects: 977, done.
remote: Counting objects: 100% (553/553), done.
remote: Compressing objects: 100% (217/217), done.
remote: Total 977 (delta 418), reused 445 (delta 336), pack-reused 424
Receiving objects: 100% (977/977), 223.83 KiB | 690.00 KiB/s, done.
Resolving deltas: 100% (674/674), completed with 38 local objects.
From https://github.com/fukamachi/qlot
   caaa359..cf2d1c7  master                      -> origin/master
 * [new branch]      1.3                         -> origin/1.3
 * [new branch]      1.3-bsd-tar                 -> origin/1.3-bsd-tar
 * [new branch]      add-tests                   -> origin/add-tests
 * [new branch]      change-run-lisp-arg         -> origin/change-run-lisp-arg
 * [new branch]      disable-color               -> origin/disable-color
 * [new branch]      error-print                 -> origin/error-print
 * [new branch]      fix/qlot-fetch-quietly      -> origin/fix/qlot-fetch-quietly
 * [new branch]      fix/qlot-installer          -> origin/fix/qlot-installer
 * [new branch]      mtstickney-symlink_handling -> origin/mtstickney-symlink_handling
 * [new branch]      no-dexador-feature          -> origin/no-dexador-feature
 * [new branch]      parallel-install            -> origin/parallel-install
 * [new branch]      refactor-distify            -> origin/refactor-distify
 * [new branch]      refactor-utils-asdf         -> origin/refactor-utils-asdf
 * [new branch]      tweak-https-fetch           -> origin/tweak-https-fetch
 * [new tag]         1.3.12                      -> 1.3.12
 * [new tag]         1.3.10                      -> 1.3.10
 * [new tag]         1.3.11                      -> 1.3.11
 * [new tag]         1.3.4                       -> 1.3.4
 * [new tag]         1.3.5                       -> 1.3.5
 * [new tag]         1.3.6                       -> 1.3.6
 * [new tag]         1.3.7                       -> 1.3.7
 * [new tag]         1.3.8                       -> 1.3.8
 * [new tag]         1.3.9                       -> 1.3.9
 * [new tag]         1.4.0                       -> 1.4.0
Updating caaa359..cf2d1c7
Fast-forward
 .github/ISSUE_TEMPLATE/bug_report.md               |  20 +++
 .github/workflows/ci-windows.yaml                  |   2 +-
 .github/workflows/ci.yaml                          |   4 +
 .github/workflows/release.yaml                     |   5 +-
 Makefile                                           |  26 ++++
 README.markdown                                    | 128 ++++++++++---------
 local-init/{https.lisp => qlot-10-https.lisp}      |  18 ++-
 local-init/{qlot-setup.lisp => qlot-99-setup.lisp} |   4 +-
 qlot-tests.asd                                     |   3 +-
 qlot.asd                                           |   7 +-
 scripts/fetch.sh                                   |  26 +++-
 scripts/install.sh                                 |  45 +++++++
 scripts/qlot                                       |  27 ----
 scripts/qlot-installer.sh                          |  72 ++++++-----
 scripts/qlot-uninstaller.sh                        |  48 ++++---
 scripts/run.sh                                     |  58 ++++-----
 scripts/setup.sh                                   |  27 ++--
 src/add.lisp                                       |  40 ++++--
 src/bundle.lisp                                    |  52 ++------
 src/check.lisp                                     | 169 +++++++++++++++++++++++++
 src/cli.lisp                                       | 993 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++---------------------------------------------------
 src/errors.lisp                                    |  34 +++++
 src/init.lisp                                      |  62 +++++++++
 src/install.lisp                                   | 400 ++++++++++++++++++++++-------------------------------------
 src/install/quicklisp.lisp                         |  31 ++++-
 src/logger.lisp                                    |  20 +--
 src/main.lisp                                      |  65 +++++++++-
 src/parser.lisp                                    |  21 ++--
 src/proxy.lisp                                     |   4 +-
 src/secure-downloader.lisp                         |  97 +++------------
 src/server.lisp                                    |  80 +++---------
 src/source/base.lisp                               |  19 ++-
 src/source/dist.lisp                               |  43 +++++--
 src/source/github.lisp                             |  55 ++++----
 src/source/ql.lisp                                 |  13 +-
 src/utils.lisp                                     |  25 +++-
 src/utils/asdf.lisp                                |  79 +++++++-----
 src/utils/cli.lisp                                 |  32 ++---
 src/utils/dependencies.lisp                        |  31 +++++
 src/utils/distify.lisp                             |  34 ++---
 src/utils/git.lisp                                 |  10 +-
 src/utils/project.lisp                             | 144 +++++++++++++++------
 src/utils/qlot.lisp                                |  38 ++++++
 src/utils/shell.lisp                               |  60 +++++----
 tests/distify/dist.lisp                            |   5 +-
 tests/distify/github.lisp                          |   4 +-
 tests/install.lisp                                 |  44 +++++--
 tests/install/quicklisp.lisp                       |   4 +-
 tests/main.lisp                                    |   2 +-
 tests/run-tests.sh                                 |   4 +-
 tests/utils/asdf.lisp                              |  19 +++
 51 files changed, 2007 insertions(+), 1246 deletions(-)
 create mode 100644 .github/ISSUE_TEMPLATE/bug_report.md
 create mode 100644 Makefile
 rename local-init/{https.lisp => qlot-10-https.lisp} (80%)
 rename local-init/{qlot-setup.lisp => qlot-99-setup.lisp} (89%)
 create mode 100755 scripts/install.sh
 delete mode 100755 scripts/qlot
 create mode 100644 src/check.lisp
 create mode 100644 src/init.lisp
 create mode 100644 src/utils/dependencies.lisp
 create mode 100644 src/utils/qlot.lisp
 create mode 100644 tests/utils/asdf.lisp
[1/3] System 'qlot' found. Loading the system..
[2/3] Processing build-hook..
[3/3] Attempting to install the scripts in roswell/ subdirectory of the system...
/Users/username/.roswell/bin/qlot

$ qlot --version
Qlot 1.2.11
```

It looks like it git pulled correctly since it lists the tags and latest commits, but for some reason the version stayed the same. I even opened a new terminal window to check with `qlot --version` and got the same result.

What do you think?

**Comment by daninus14** [comment]
just for reference:
```
$ ros --version
roswell 23.10.14.114(NO-GIT-REVISION)
```

**Comment by fukamachi** [comment]
`ros install qlot` installs it from the Quicklisp dist, which hasn't been released since October.
Try `ros install fukamachi/qlot` to install it from GitHub instead.

**Comment by daninus14** [comment]
ok thanks, just did. Thanks for the help.

It didn't work at first, I just realized why, and here I will explain what else happened:

I had two installations of qlot: one with roswell, and one with the install script.

`which qlot` and `where qlot` both returned `~/.roswell/bin/qlot`.

However when running `ros run -Q -L sbcl-bin` and starting the REPL, the function `(ql:where-is-system "qlot")` would return the `~/.qlot/qlot`.

So even though the shell was calling the `qlot` script stored in `~/.roswell/bin/qlot`, since that script just loads qlot in lisp, and quicklisp, the version available in quicklisp in the REPL is loaded.

This means that even though the directory of that script is in `~/.roswell/bin/qlot` and the git repo for roswell is `~/.roswell/local-projects/fukamachi/qlot`, quicklisp loads the repo in `~/.qlot/qlot`.

I solved it by running the uninstaller script for qlot to remove the `~/.qlot` install, and now it defaults to the one in roswell, which is what quicklisp loads.

It's unclear to me why quicklisp loaded the home based qlot repository instead of the roswell based one. It could be because it's provided first in a list of directories to check, and since the system was already defined, it loaded the first definition and ignored the second system definition.

Thanks for the help, I don't think I would've figured it out without trying to install the github repo version like you adviced.

Not sure if this will be helpful for anyone or if we should add it to the docs somehow, but I don't really know what. Maybe a troubleshooting qlot section which links to all the issues and things that have come up?

I'm closing the issue now since it works, thanks.
